main.tf

```
@@ -171,14 +171,30 @@ resource "aws_codedeploy_deployment_group" "default" {
     }
   }
 
-  dynamic "ec2_tag_set" {
+  # Note that you cannot have both ec_tag_filter and ec2_tag_set vars set!
+  # See https://docs.aws.amazon.com/cli/latest/reference/deploy/create-deployment-group.html for details
+  dynamic "ec2_tag_filter" {
     for_each = var.ec2_tag_filter == null ? [] : var.ec2_tag_filter
```

the default value is an empty list so that should be checked

```suggestion
    for_each = length(var.ec2_tag_filter) > 0 ? [] : var.ec2_tag_filter
```